### PR TITLE
Replaced laminas-mail with symfony/mailer

### LINKED ISSUE #60: replace laminas-mail ?
since laminas mail is in security only mode 
should we repalce it with symfony/mailer ?

NOTE
> laminas-mail is deprecated and discontinued
>  can be used in production temporary
> need to be replaced with the recommended replacement

### PR DESCRIPTION
**Pull request by SergiuBota1**
Replaced [laminas-mail](https://github.com/laminas/laminas-mail) with [symfony/mailer](https://github.com/symfony/mailer)  #60 
Bumped to php 8.4  #63

### CODE REVIEW

File: docs/book/v5/configuration.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Configuration
+
+Register `dot-mail` in you project by adding `Dot\Mail\ConfigProvider::class` to your configuration aggregator (to `config/config.php` for example).
+After registering the `ConfigProvider` load the configuration file (`config/mail.global.php.dist`) by removing it's `.dist` extension and adding it to the `config/autoload` folder.
```
Comment (alexmerlin) [inline docs/book/v5/configuration.md]:
```suggestion
After registering the `ConfigProvider` copy the configuration file `config/mail.global.php.dist` into your project's `config/autoload/` directory as `mail.global.php`.
```

File: docs/book/v5/configuration.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+## Transport configuration
+
+`dot-mail` uses the `transport` key under the main `dot_mail` configuration key to select the email transport.
+It has four email transport classes available by default (`SmtpTransport`), one of which is to be added under the `dot_mail.transport` key for use.
+
+Sending email with the `Smtp` transport requires valid data for the values under `dot-mail.default.smtp_options`, which is only used in this case.
+
+- Note: the configured path must be a writable directory
```
Comment (alexmerlin) [inline docs/book/v5/configuration.md]:
```suggestion
> The configured path must be a writable directory
```

File: docs/book/v5/configuration.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+```php
+'file_options' => [
+        'path' => 'data/mail/output',
```
Comment (alexmerlin) [inline docs/book/v5/configuration.md]:
```suggestion
    'path' => 'data/mail/output',
```

File: docs/book/v5/configuration.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        //'callback' => null,
```
Comment (alexmerlin) [inline docs/book/v5/configuration.md]:
```suggestion
    //'callback' => null,
```

File: docs/book/v5/transports.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Transports
+
+`dot-mail` can use any transport class that implements `symfony\Component\Mailer\Transport\TransportInterface`, with the standard transport available being:
+
+- `Laminas\Mail\Transport\Smtp`
+
+- Note: feel free to use any custom transport you desire, provided it implements the mentioned `TransportInterface`.
```
Comment (alexmerlin) [inline docs/book/v5/transports.md]:
```suggestion
> Feel free to use any custom transport you desire, provided it implements the mentioned `TransportInterface`.
```

File: docs/book/v5/usage.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+```php
+$result = $this->mailService->send();
+if (! $result->isValid()) {
+    //log the error
+    error_log($result->getMessage());
+}
+```
+
+**Note : Invalid e-mail messages will not be sent.**
```
Comment (alexmerlin) [inline docs/book/v5/usage.md]:
```suggestion
> Invalid e-mail messages will not be sent.
```

File: mkdocs.yml
Comment (alexmerlin) [inline mkdocs.yml:1]: Add `v5` documentation files to `mkdocs.yml`.

**Review by alexmerlin — CHANGES_REQUESTED** [review]

File: composer.json [hunk 7]
```diff
@@ -1,21 +1,21 @@
...
     "name": "dotkernel/dot-mail",
     "type": "library",
-    "description": "DotKernel mail component based on laminas-mail",
+    "description": "Dotkernel mail component based on symfony mailer",
     "license": "MIT",
     "homepage": "https://github.com/dotkernel/dot-mail",
     "keywords": [
         "mail",
         "services",
-        "laminas",
```
Comment (arhimede) [inline composer.json:10]: keep laminas keyword

File: docs/book/v5/transports.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+`Smtp` connects to the configured SMTP host in order to handle sending emails.
+
+- As the email is not sent, this transport can be helpful in development, with the access to the message being potentially useful in tests as well
+
+```php
+$this->mailService->setBody('First email body');
+$this->mailService->setSubject('First email subject');
+$this->mailService->getMessage()->setTo('email@example.com');
+// The email is not sent to the email, instead it is stored in memory
```
Comment (arhimede) [inline docs/book/v5/transports.md]: why in memory ? what is the meaning of this comment ?

File: docs/book/v5/transports.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+$this->mailService->send();
+
+$this->mailService->setBody('Second email body');
+$this->mailService->setSubject('Second email subject');
+$this->mailService->getMessage()->setTo('email@example.com');
+
+// The email is not sent to the email either, it overwrites the previously "sent" email in memory
```
Comment (arhimede) [inline docs/book/v5/transports.md]: no memory

File: docs/book/v5/transports.md (hunk 5 above)
Comment (arhimede) [reply docs/book/v5/transports.md]: Please clarify that only SENDMAIL and SMTP are used.

File: .github/workflows/codecov.yml [hunk 10]
```diff
@@ -15,9 +15,9 @@ jobs:
           - ubuntu-latest
 
         php:
-          - "8.1"
           - "8.2"
           - "8.3"
+          - "8.4"
```
Comment (arhimede) [inline .github/workflows/codecov.yml]: remove this one for the moment

**Review by arhimede — APPROVED** [review]

**Pull request merged**